datasets/il/wmd_banned/crawler.py

```diff
@@ -0,0 +1,244 @@
+        date_str = re.sub("ב", "", date_str, flags=re.IGNORECASE)
+
+    date_fmt = ["%d %B %Y", "%d %B. %Y"]
+    for fmt in date_fmt:
+        try:
+            date_obj = datetime.datetime.strptime(date_str, fmt)
+            return date_obj.date()
+        except ValueError:
+            pass
+
+
+def extract_passport_no(text: str):
+    """
+    Extract passport numbers from a given text.
+    """
+    if not text:
+        return None
+    text = str(text)
+    pattern = r"\b[A-Z0-9]{5,}\b"
+    matches = re.findall(pattern, text)
+
+    return matches
+
+
+def extract_n_pop_address(text: str):
+    """
+    Extract address and update the text by removing the extracted address.
+    """
+    if not text:
+        return None, None
+
+    pattern = r"(address|location):\s*(.*)"
+    match = re.search(pattern, text, re.DOTALL | re.IGNORECASE)
+    if match:
+        address = match.group(2).strip()
+
+        # Remove the extracted location from the original text
+        updated_text = re.sub(
+            pattern, "", text, flags=re.DOTALL | re.IGNORECASE
+        ).strip()
+
+        return address, updated_text
+
+    else:
+        return None, text
+
+
+def clean_name(name: str | List[str]):
+    """
+    Clean the given name (or list of names) by removing newlines, quotes, colons, and trailing dots
+    """
+    if isinstance(name, list):
+        return [re.sub(r"[\n':]|^\.*|\.*$", "", match).strip() for match in name]
+
+    return re.sub(r"[\n':]|^\.*|\.*$", "", name).strip()
+
+
+def format_numbered_listings(text: str):
+    """
+    Extract items from a text that follows numbered or lettered listings.
+    """
+    if not text:
+        return None
+    patterns = [
+        r"\b[a-z]\)\s(.*?)(?=\s[a-z]\)|$)",  # Matches 'a) text b) text'
+        r"\b\d+:\s(.*?)(?=\s\d+:|$)",  # Matches '1: text 2: text'
+    ]
+    for pattern in patterns:
+        matches = re.findall(pattern, text, re.DOTALL | re.VERBOSE | re.IGNORECASE)
+        if matches:
+            return matches
+
+    else:
+        return text
+
+
+def sheet_to_tuple(sheet):
+    """
+    Convert rows from an Excel sheet to a tuple.
+    Index of tuple is used to locate attributes in the crawler
+    as the headers/keys are in hebrew
+    """
+    headers: Optional[List[str]] = None
+    for row in sheet.iter_rows(min_row=0):
+        cells = [c.value for c in row]
+
+        if headers is None:
+            if all(cells):
+                headers = [h for h in cells]
+            continue
+
+        if len(cells) == len(headers):
+            if 'מס"ד' in cells:  # cell that contains headers, skip it
+                continue
+
+            none_count = sum(1 for item in cells if not item)
+            if (
+                none_count > len(headers) // 2
+            ):  #  cells that do not contain data up to half of the headers count, skip it as it is probably a subheader
+                continue
+
+            yield cells
```

Comment:
this was a good idea, but could you see if you can use [h.parse_xlsx_table](https://zavod.opensanctions.org/helpers/#zavod.helpers.parse_xlsx_sheet) with the column translation lookup argument and skip repeated header rows in the item loop?

https://zavod.opensanctions.org/best_practices/datapatch_lookups/#translate-headers-using-datapatch-lookups